### PR TITLE
Add the instruction on how to use spec.version and spec.manifests

### LINKED ISSUE #2768: Documentation about how to install customized Knative manifests
**Describe the change you'd like to see**
Operator has implements two new fields: spec.version and spec.manifests to install customized manifests. We need the documentation about how to use them.

### PR DESCRIPTION
**Pull request by houshengbo**
Fixes #2768 
## Proposed Changes 

- This PR added the explanation on how to install future Knative releases and customized knative manifests with the operator.

### CODE REVIEW

File: docs/install/knative-with-operators.md [hunk 1]
```diff
@@ -82,8 +82,34 @@ kubectl logs -f deploy/knative-operator
...
+ name: knative-serving
+---
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+```
+
+If you do not specify the version with `spec.version`, the operator will take the latest one available in the operator.
```
Comment (mattmoor) [inline docs/install/knative-with-operators.md:104]: Remove trailing whitespace:
```suggestion
If you do not specify the version with `spec.version`, the operator will take the latest one available in the operator.
```

File: docs/install/knative-with-operators.md [hunk 2]
```diff
@@ -132,10 +208,36 @@ knative-serving   <version number>    True
...
+ name: knative-eventing
+---
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeEventing
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+```
+
+If you do not specify the version with `spec.version`, the operator will take the latest one available in the operator.
```
Comment (mattmoor) [inline docs/install/knative-with-operators.md:232]: Remove trailing whitespace:
```suggestion
If you do not specify the version with `spec.version`, the operator will take the latest one available in the operator.
```

File: docs/install/knative-with-operators.md [hunk 3]
```diff
@@ -82,8 +82,34 @@ kubectl logs -f deploy/knative-operator
...
+ name: knative-serving
+---
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+```
+
+If you do not specify the version with `spec.version`, the operator will take the latest one available in the operator.
```
Comment (mpetason) [inline docs/install/knative-with-operators.md:104]:
```suggestion
If you do not specify the version with `spec.version`, the operator default to the latest available version.
```

File: docs/install/knative-with-operators.md [hunk 4]
```diff
@@ -82,8 +82,34 @@ kubectl logs -f deploy/knative-operator
...
+
+{{< /tab >}}
+
+{{% tab name="Install Future Knative Serving" %}} You do not need to upgrade the operator to a newer version, to install
+the new releases of Knative Serving. If Knative Serving launches a new version, e.g. `ma.mi.pa`, you can install it by
```
Comment (mpetason) [inline docs/install/knative-with-operators.md:109]:
```suggestion
{{% tab name="Install Future Knative Serving" %}} You do not need to upgrade the operator to a newer version to install new releases of Knative Serving. If Knative Serving launches a new version, e.g. `ma.mi.pa`, you can install it by
```

File: docs/install/knative-with-operators.md [hunk 5]
```diff
@@ -94,9 +120,59 @@ kind: KnativeServing
...
+    - URL: https://github.com/knative/serving/releases/download/v{VERSION}/serving-post-install-jobs.yaml
+    - URL: https://github.com/knative-sandbox/net-istio/releases/download/v0.17.0/net-istio.yaml
 ```
 
+The field `spec.version` is used to set the version of Knative Serving. Replace `ma.mi.pa` with the correct version number.
+The tag `${VERSION}` is automatically replaced with the version number from `spec.version` by the operator.
+
+The field `spec.manifests` is used to specify one or multiple URL links of Knative Serving component. Do not forget to
+add the valid URL of the Knative network ingress plugin. Knative Serving component is still tightly-coupled with a network
+ingress plugin in the operator. As in the above example, you can use `net-istio`. The list of URLs is order critical. Put
```
Comment (mpetason) [inline docs/install/knative-with-operators.md:137]:
```suggestion
ingress plugin in the operator. As in the above example, you can use `net-istio`. The ordering of the URLs is critical. Put
```

File: docs/install/knative-with-operators.md [hunk 6]
```diff
@@ -94,9 +120,59 @@ kind: KnativeServing
...
+spec:
+  version: 0.17.0
+  manifests:
+    - URL: https://my-serving/serving.yaml
+    - URL: https://github.com/knative-sandbox/net-istio/releases/download/v0.17.0/net-istio.yaml
+```
+
+You can make the customized Knative Serving available in one or multiple links, as the `spec.manifests` supports a list
+of links. The list of URLs is order critical. Put the manifest you want to apply first on the top. We strongly recommend
+you to specify the version and the valid links to the customized Knative Serving, by leveraging both of `spec.version`
```
Comment (mpetason) [inline docs/install/knative-with-operators.md:169]:
```suggestion
you to specify the version and the valid links to the customized Knative Serving, by leveraging both `spec.version`
```

File: docs/install/knative-with-operators.md (hunk 3 above)
Comment (abrennan89) [reply docs/install/knative-with-operators.md:104]: "If you do not specify a version by using `spec.version`, the operator defaults to the latest available version."

File: docs/install/knative-with-operators.md [hunk 7]
```diff
@@ -82,8 +82,34 @@ kubectl logs -f deploy/knative-operator
...
+  namespace: knative-serving
+```
+
+If you do not specify the version with `spec.version`, the operator will take the latest one available in the operator.
+
+{{< /tab >}}
+
+{{% tab name="Install Future Knative Serving" %}} You do not need to upgrade the operator to a newer version, to install
+the new releases of Knative Serving. If Knative Serving launches a new version, e.g. `ma.mi.pa`, you can install it by
+applying the following content:
```
Comment (abrennan89) [inline docs/install/knative-with-operators.md:110]:
```suggestion
applying a YAML file containing the following:
```

File: docs/install/knative-with-operators.md [hunk 8]
```diff
@@ -94,9 +120,59 @@ kind: KnativeServing
...
   name: knative-serving
   namespace: knative-serving
-EOF
+spec:
+  version: ma.mi.pa
+  manifests:
+    - URL: https://github.com/knative/serving/releases/download/v${VERSION}/serving-core.yaml
+    - URL: https://github.com/knative/serving/releases/download/v{VERSION}/serving-hpa.yaml
+    - URL: https://github.com/knative/serving/releases/download/v{VERSION}/serving-post-install-jobs.yaml
+    - URL: https://github.com/knative-sandbox/net-istio/releases/download/v0.17.0/net-istio.yaml
```
Comment (abrennan89) [inline docs/install/knative-with-operators.md:129]: Should we change `v0.17.0` to `v${VERSION}` too?

File: docs/install/knative-with-operators.md [hunk 9]
```diff
@@ -94,9 +120,59 @@ kind: KnativeServing
...
+
+{{< /tab >}}
+
+{{% tab name="Install Customized Knative Serving" %}} The operator provides you the flexibility to install customized
+Knative Serving based your own requirements. As long as the manifests of customized Knative Serving are accessible to
+the operator, they can be installed.
+
+For example, the version of the customized Knative Serving is 0.17.0, and it is available at `https://my-serving/serving.yaml`.
+You can create the content of Serving CR as below:
+
```
Comment (abrennan89) [inline docs/install/knative-with-operators.md:148]: I'd rewrite any of this to make it general, and if we can always just refer to a generic version or variable for it instead of saying specific number versions, e.g. v.0.17.0. It just saves maintenance / updates later.

File: docs/install/knative-with-operators.md [hunk 10]
```diff
@@ -146,9 +248,55 @@ kind: KnativeEventing
...
+metadata:
+ name: knative-eventing
+---
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeEventing
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+spec:
+  version: 0.17.0
```
Comment (abrennan89) [inline docs/install/knative-with-operators.md:286]: Same comments pretty much for this as above.

File: docs/install/knative-with-operators.md (hunk 8 above)
Comment (houshengbo) [reply docs/install/knative-with-operators.md:129]: I would rather keep v0.17.0 for net-istio, because spec.version is for the version of knative.
net-istio is ingress plugin. it could be any supported version.

**Comment by houshengbo** [comment]
@mpetason @abrennan89 I have changed this patch based on the comments. Thx!

File: docs/install/knative-with-operators.md [hunk 11]
```diff
@@ -94,9 +120,60 @@ kind: KnativeServing
...
+ingress plugin in the operator. As in the above example, you can use `net-istio`. The ordering of the URLs is critical.
+Put the manifest you want to apply first on the top.
+
+{{< /tab >}}
+
+{{% tab name="Install Customized Knative Serving" %}} The operator provides you the flexibility to install customized
+Knative Serving based your own requirements. As long as the manifests of customized Knative Serving are accessible to
+the operator, they can be installed.
+
+For example, the version of the customized Knative Serving is `ma.mi.pa`, and it is available at `https://my-serving/serving.yaml`.
```
Comment (mpetason) [inline docs/install/knative-with-operators.md:146]: @evankanderson What could we use here instead of `ma.mi.pa` - I don't see it used anywhere else in docs. Is there a standard we use for an example version, or just the latest version.

Comment (evankanderson) [reply docs/install/knative-with-operators.md:146]: I would probably use `{{spec.version}}` here if possible, or a concrete version like `0.16.4`. The major.minor.patch as two-letter abbreviations seems a bit non-standard.

Comment (houshengbo) [reply docs/install/knative-with-operators.md]: I have replace all the existences of ma.mi.pa with {{spec.version}}.

File: docs/install/knative-with-operators.md [hunk 12]
```diff
@@ -94,9 +120,60 @@ kind: KnativeServing
...
+    - URL: https://github.com/knative/serving/releases/download/v{VERSION}/serving-post-install-jobs.yaml
+    - URL: https://github.com/knative/net-istio/releases/download/v0.17.0/net-istio.yaml
 ```
 
+The field `spec.version` is used to set the version of Knative Serving. Replace `{{spec.version}}` with the correct version number.
+The tag `${VERSION}` is automatically replaced with the version number from `spec.version` by the operator.
+
+The field `spec.manifests` is used to specify one or multiple URL links of Knative Serving component. Do not forget to
+add the valid URL of the Knative network ingress plugin. Knative Serving component is still tightly-coupled with a network
+ingress plugin in the operator. As in the above example, you can use `net-istio`. The ordering of the URLs is critical.
```
Comment (nak3) [inline docs/install/knative-with-operators.md:137]: It sounds like uses can choose other ingress plugin. 
But only `net-istio` is supported now? (I assumed it because `Before you begin` section says `Download and Istio`.

Comment (houshengbo) [reply docs/install/knative-with-operators.md:137]: that's tricky. since we can specify the plugin, we technically can install any of them with the customized yaml.

File: docs/install/knative-with-operators.md [hunk 13]
```diff
@@ -82,8 +82,34 @@ kubectl logs -f deploy/knative-operator
...
+  name: knative-serving
+  namespace: knative-serving
+```
+
+If you do not specify a version by using spec.version, the operator defaults to the latest available version.
+
+{{< /tab >}}
+
+{{% tab name="Install Future Knative Serving" %}} You do not need to upgrade the operator to a newer version to install
+new releases of Knative Serving. If Knative Serving launches a new version, e.g. `ma.mi.pa`, you can install it by
```
Comment (evankanderson) [inline docs/install/knative-with-operators.md:109]: `ma.mi.pa` here.

File: docs/install/knative-with-operators.md [hunk 14]
```diff
@@ -94,9 +120,60 @@ kind: KnativeServing
...
   name: knative-serving
   namespace: knative-serving
-EOF
+spec:
+  version: {{spec.version}}
+  manifests:
+    - URL: https://github.com/knative/serving/releases/download/v${VERSION}/serving-core.yaml
+    - URL: https://github.com/knative/serving/releases/download/v{VERSION}/serving-hpa.yaml
+    - URL: https://github.com/knative/serving/releases/download/v{VERSION}/serving-post-install-jobs.yaml
+    - URL: https://github.com/knative/net-istio/releases/download/v0.17.0/net-istio.yaml
```
Comment (evankanderson) [inline docs/install/knative-with-operators.md:129]: Strange not to have a {VERSION} here, but I'm assuming it's because there might be a different patch version.

File: docs/install/knative-with-operators.md [hunk 15]
```diff
@@ -94,9 +120,60 @@ kind: KnativeServing
...
 ```
 
+The field `spec.version` is used to set the version of Knative Serving. Replace `{{spec.version}}` with the correct version number.
+The tag `${VERSION}` is automatically replaced with the version number from `spec.version` by the operator.
```
Comment (evankanderson) [inline docs/install/knative-with-operators.md:133]: This says `${VERSION}`, but the example uses `v{VERSION}` with no `$`.

Comment (evankanderson) [reply docs/install/knative-with-operators.md:133]: (Or, if this is talking about a different concept, it's not clear what it references.)

File: docs/install/knative-with-operators.md (hunk 12 above)
Comment (evankanderson) [reply docs/install/knative-with-operators.md:137]: We don't have official support for non-Istio installs yet, but bring-your-own YAML provides an out.

File: docs/install/knative-with-operators.md (hunk 14 above)
Comment (evankanderson) [reply docs/install/knative-with-operators.md:129]: For a future update, we should figure out how to unify these and give some additional hints on how versioning should work.

**Review by evankanderson — COMMENTED** [review]
/approve

One missed replacement, and a bit of confusion on tag vs string replacement, but I think if you fix the first comment (the `ma.mi.pa`), this `/lgtm`.

**Review by evankanderson — COMMENTED** [review]
/lgtm

File: docs/install/knative-with-operators.md (hunk 12 above)
Comment (nak3) [reply docs/install/knative-with-operators.md:137]: I understand. Thank you.

**Comment by nak3** [comment]
Hi @houshengbo @evankanderson 
The change https://github.com/knative/operator/commit/c7a16686e8114ff5592fa2bddc65b9dac5b0efb5 was merged to 0.17 so this PR also needs to be backport to 0.17?

cc @FWiesner
